### DISCUSSION TITLE
[@types/react] Add support to render an array of elements in FunctionComponent

### DISCUSSION THREAD

**Discussion by navarroaxel** [DefinitelyTyped/DefinitelyTyped#57430] [Issues with a @types package]
The official blog post to announce React v16.0.0 says [`you can now return an array of elements from a component`]( https://reactjs.org/blog/2017/09/26/react-v16.0.html#new-render-return-types-fragments-and-strings) without wrap it in a `<Fragment/>`.

The following code:

```javascript
const arr = [1, 2, 3];

const MyList: FC = () => arr.map(item => <li key={item}>{item}</li>);
```

give this error:
> TS2739: Type 'Element[]' is missing the following properties from type 'ReactElement<any, any>': type, props, key

This error could be fixed if we add `ReactElement<any, any>[]` to the `FunctionComponent` interface:
```
interface FunctionComponent<P = {}> {
    (props: PropsWithChildren<P>, context?: any): ReactElement<any, any> | ReactElement<any, any>[] | null;
    propTypes?: WeakValidationMap<P> | undefined;
    contextTypes?: ValidationMap<any> | undefined;
    defaultProps?: Partial<P> | undefined;
    displayName?: string | undefined;
}
```

Also, the supported return types includes `string`, `number`, `boolean`, but the package `@types/react` doesn't support them. https://reactjs.org/docs/react-component.html#render

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
Duplicate of https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18912,  https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18051
